File: Python/Software Development Module - Python/Revision/Exam2018/InterestCalculator/InterestCalculator.py
"""
InterestCalculator.py
Billy Cussen
02/11/2020
"""

class myInterestCalculator:

    def __init__(self):
        self.interestTotal = 0.0

    def getInterest(self, amount, months, accessMoney):
        if months == 6:  
            if accessMoney:
                self.interestTotal = amount * 0.0002
            else:
                self.interestTotal = amount * 0.0005
        elif months == 12:
            if accessMoney:
                self.interestTotal = amount * 0.0015
            else:
                self.interestTotal = amount * 0.002
        elif months == 18:
            if accessMoney:
                self.interestTotal = amount * 0.002
            else:
                self.interestTotal = amount * 0.004
        else:
            raise Exception
        return self.interestTotal

    def getHighestInterest(self, allDeposits):
        highestInterest = allDeposits[0]
        for d in allDeposits:
            if d > highestInterest:
                highestInterest = d
        return highestInterest

File: Java/Software Development Module - Java/Revision/Exam2018/InterestCalculator.java
/*
InterestCalculator.java
Billy Cussen
03/09/2020
*/

import java.util.InputMismatchException;

public class InterestCalculator{

    //Data Members
    private double interestTotal;
    private double highestInterest;

    //Constructor
    public InterestCalculator(){
    }

    //Compute, Setter & Getter
    public double getInterest (double amount, int months, boolean accessMoney) throws InputMismatchException{
        switch(months){
            case 6:
                if(accessMoney){
                    this.interestTotal = amount * 0.0002;
                }else{
                    this.interestTotal = amount * 0.0005;
                }
                break;
            case 12:
                if(accessMoney){
                    this.interestTotal = amount * 0.0015;
                }else{
                    this.interestTotal = amount * 0.002;
                }
                break;
            case 18:
                if(accessMoney){
                    this.interestTotal = amount * 0.002;
                }else{
                    this.interestTotal = amount * 0.004;
                }
                break;
            default:
                throw new InputMismatchException("Something went wrong here!");
        }
        return this.interestTotal;
    }

    public double getHighestInterest(double[] allInterest){
        this.highestInterest = allInterest[0];
        for(int i = 1; i < allInterest.length; i++){
            if(allInterest[i] >= this.highestInterest){
                this.highestInterest = allInterest[i];
            }
        }
        return this.highestInterest;
    }

}